internal/sbi/server.go

```diff
@@ -0,0 +1,199 @@
+	bdtPolicyGroup.Use(func(c *gin.Context) {
+		bdtRouterAuthorizationCheck.Check(c, pcf_context.GetSelf())
+	})
+	applyRoutes(bdtPolicyGroup, bdtPolicyRoutes)
+
+	httpcallbackRoutes := s.getHttpCallBackRoutes()
+	httpcallbackGroup := s.router.Group(factory.PcfCallbackResUriPrefix)
+	applyRoutes(httpcallbackGroup, httpcallbackRoutes)
+
+	oamRoutes := s.getOamRoutes()
+	oamGroup := s.router.Group(factory.PcfOamResUriPrefix)
+	oamRouterAuthorizationCheck := util.NewRouterAuthorizationCheck(models.ServiceName_NPCF_OAM)
+	oamGroup.Use(func(c *gin.Context) {
+		oamRouterAuthorizationCheck.Check(c, pcf_context.GetSelf())
+	})
+	applyRoutes(oamGroup, oamRoutes)
+
+	policyAuthorizationRoutes := s.getPolicyAuthorizationRoutes()
+	policyAuthorizationGroup := s.router.Group(factory.PcfPolicyAuthResUriPrefix)
+	policyAuthorizationRouterAuthorizationCheck := util.
+		NewRouterAuthorizationCheck(models.ServiceName_NPCF_POLICYAUTHORIZATION)
+	policyAuthorizationGroup.Use(func(c *gin.Context) {
+		policyAuthorizationRouterAuthorizationCheck.Check(c, pcf_context.GetSelf())
+	})
+	applyRoutes(policyAuthorizationGroup, policyAuthorizationRoutes)
+
+	uePolicyRoutes := s.getUePolicyRoutes()
+	uePolicyGroup := s.router.Group(factory.PcfUePolicyCtlResUriPrefix)
+	applyRoutes(uePolicyGroup, uePolicyRoutes)
+
+	s.router.Use(cors.New(cors.Config{
+		AllowMethods: []string{"GET", "POST", "OPTIONS", "PUT", "PATCH", "DELETE"},
+		AllowHeaders: []string{
+			"Origin", "Content-Length", "Content-Type", "User-Agent",
+			"Referrer", "Host", "Token", "X-Requested-With",
+		},
+		ExposeHeaders:    []string{"Content-Length"},
+		AllowCredentials: true,
+		AllowAllOrigins:  true,
+		MaxAge:           CorsConfigMaxAge,
+	}))
+
+	cfg := s.Config()
+	bindAddr := cfg.GetSbiBindingAddr()
+	logger.SBILog.Infof("Binding addr: [%s]", bindAddr)
+	var err error
+	if s.httpServer, err = httpwrapper.NewHttp2Server(bindAddr, tlsKeyLogPath, s.router); err != nil {
+		logger.InitLog.Errorf("Initialize HTTP server failed: %v", err)
+		return nil, err
+	}
+	s.httpServer.ErrorLog = log.New(logger.SBILog.WriterLevel(logrus.ErrorLevel), "HTTP2: ", 0)
+
+	return s, nil
+}
+
+func (s *Server) Run(traceCtx context.Context, wg *sync.WaitGroup) error {
+	var err error
+	_, s.Context().NfId, err = s.Consumer().SendRegisterNFInstance(context.Background())
+	if err != nil {
+		logger.InitLog.Errorf("CHF register to NRF Error[%s]", err.Error())
+	}
+
+	wg.Add(1)
+	go s.startServer(wg)
+
+	return nil
+}
+
+func (s *Server) Stop(traceCtx context.Context) {
+	const defaultShutdownTimeout time.Duration = 2 * time.Second
+
+	if s.httpServer != nil {
+		logger.SBILog.Infof("Stop SBI server (listen on %s)", s.httpServer.Addr)
+		toCtx, cancel := context.WithTimeout(context.Background(), defaultShutdownTimeout)
+		defer cancel()
+		if err := s.httpServer.Shutdown(toCtx); err != nil {
+			logger.SBILog.Errorf("Could not close SBI server: %#v", err)
+		}
+	}
+}
```

Comment:
Align naming convention with `http.Server.Shutdown()`.

```suggestion
func (s *Server) Shutdown(traceCtx context.Context) {
	const defaultShutdownTimeout time.Duration = 2 * time.Second

	if s.httpServer != nil {
		logger.SBILog.Infof("Stop SBI server (listen on %s)", s.httpServer.Addr)
		toCtx, cancel := context.WithTimeout(context.Background(), defaultShutdownTimeout)
		defer cancel()
		if err := s.httpServer.Shutdown(toCtx); err != nil {
			logger.SBILog.Errorf("Could not close SBI server: %#v", err)
		}
	}
}
```